crates/protocol/hardforks/src/ecotone.rs

```diff
@@ -105,6 +106,22 @@ impl Ecotone {
 
     /// Returns the list of [TxDeposit]s for the Ecotone network upgrade.
     pub fn deposits() -> impl Iterator<Item = TxDeposit> {
+        // Verify source has of L1Block
+        // See: <https://specs.optimism.io/protocol/ecotone/derivation.html#l1block-deployment>
+        debug_assert_eq!(
+            Self::deploy_l1_block_source(),
+            B256::from_str("0xc88a313aa75dc4fbf0b6850d9f9ae41e04243b7008cf3eadb29256d4a71c1dfd")
+                .unwrap(),
```

Comment:
Let's just use alloy's macros here so we don't even need to bring `FromStr` into scope:

```suggestion
            alloy_primitives::b256!("c88a313aa75dc4fbf0b6850d9f9ae41e04243b7008cf3eadb29256d4a71c1dfd"),
```